simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -1532,85 +1532,71 @@ auto HdMapUtils::getLateralDistance(
-  -> std::optional<double>
-{
-  if (from.lanelet_id == to.lanelet_id) {
-    if (from.s > to.s) {
-      return std::nullopt;
-    } else {
-      return to.s - from.s;
-    }
-  }
-  const auto route = getRoute(from.lanelet_id, to.lanelet_id, allow_lane_change);
-  if (route.empty()) {
-    return std::nullopt;
-  }
-  double distance = 0;
-
-  auto with_lane_change = [this](
-                            const bool allow_lane_change, const lanelet::Id current_lanelet,
-                            const lanelet::Id next_lanelet) -> bool {
-    if (allow_lane_change) {
-      auto next_lanelet_ids = getNextLaneletIds(current_lanelet);
-      auto next_lanelet_itr = std::find_if(
-        next_lanelet_ids.begin(), next_lanelet_ids.end(),
-        [next_lanelet](const lanelet::Id & id) { return id == next_lanelet; });
-      return next_lanelet_itr == next_lanelet_ids.end();
-    } else {
-      return false;
-    }
+  const traffic_simulator_msgs::msg::LaneletPose & from_pose,
+  const traffic_simulator_msgs::msg::LaneletPose & to_pose,
+  const bool allow_lane_change /*= false*/) const -> std::optional<double>
+{
+  const auto is_lane_change_required =
+    [this](const lanelet::Id current_lanelet, const lanelet::Id next_lanelet) -> bool {
+    const auto next_lanelet_ids = getNextLaneletIds(current_lanelet);
+    return std::none_of(
+      next_lanelet_ids.cbegin(), next_lanelet_ids.cend(),
+      [next_lanelet](const lanelet::Id id) { return id == next_lanelet; });
   };
 
-  /// @note in this for loop, some cases are marked by @note command. each case is explained in the document.
   /// @sa https://tier4.github.io/scenario_simulator_v2-docs/developer_guide/DistanceCalculation/
-  for (unsigned int i = 0; i < route.size(); i++) {
-    if (i < route.size() - 1 && with_lane_change(allow_lane_change, route[i], route[i + 1])) {
-      /// @note "the lanelet before the lane change" case
-      traffic_simulator_msgs::msg::LaneletPose next_lanelet_pose;
-      next_lanelet_pose.lanelet_id = route[i + 1];
-      next_lanelet_pose.s = 0.0;
-      next_lanelet_pose.offset = 0.0;
-
-      if (
-        auto next_lanelet_origin_from_current_lanelet =
-          toLaneletPose(toMapPose(next_lanelet_pose).pose, route[i], 10.0)) {
-        distance += next_lanelet_origin_from_current_lanelet->s;
-      } else {
-        traffic_simulator_msgs::msg::LaneletPose current_lanelet_pose = next_lanelet_pose;
-        current_lanelet_pose.lanelet_id = route[i];
-        if (
-          auto current_lanelet_origin_from_next_lanelet =
-            toLaneletPose(toMapPose(current_lanelet_pose).pose, route[i + 1], 10.0)) {
-          distance -= current_lanelet_origin_from_next_lanelet->s;
+  if (const auto route = getRoute(from_pose.lanelet_id, to_pose.lanelet_id, allow_lane_change);
+      not route.empty() || from_pose.lanelet_id == to_pose.lanelet_id) {
+    static constexpr double matching_distance = 10.0;
```

Comment:
Please use the @note comment to explain why the hard coded parameter is the number it is.
Even if there is no specific reason, be sure to leave a comment to that effect.

Reply:
added